### PR TITLE
Migrate create/delete/patch channel to redux

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR migrates the createChannel action to redux, removes a few unused actions and changes the use of updateChannel to patchChannel for the rename channel modal.

Affected areas:
* Create public/private channel
* Rename channel

Note: I did change the component for `ChannelPermissionGate` as it was broken in master in order to test the options in the channel dropdown menu, not sure if this is the intended fix.

@DHaussermann this should fix the issue you were seeing around channel creation that was broken in master ticket MM-12766

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12695
https://mattermost.atlassian.net/browse/MM-12766

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/permissions_gates/channel_permission_gate/index.js [hunk 1]
```diff
@@ -4,16 +4,18 @@
 import {connect} from 'react-redux';
 
 import {haveIChannelPermission} from 'mattermost-redux/selectors/entities/roles';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 
 import ChannelPermissionGate from './channel_permission_gate.jsx';
 
 function mapStateToProps(state, ownProps) {
-    if (!ownProps.channelId || ownProps.teamId === null || typeof ownProps.teamId === 'undefined') {
+    const teamId = ownProps.teamId || getCurrentTeamId(state);
```
Comment (enahum) [inline components/permissions_gates/channel_permission_gate/index.js]: can't really say if this is the proper fix

Comment (jwilander) [reply components/permissions_gates/channel_permission_gate/index.js]: @grundleborg I'm guessing we should have some other default state?

Comment (enahum) [reply components/permissions_gates/channel_permission_gate/index.js]: As I said I have no idea what the proper fix is, I just added this in the PR so the options are there to test the actions, I can remove this before merging if needed

Comment (grundleborg) [reply components/permissions_gates/channel_permission_gate/index.js]: I'm a bit out of my depth here. It's really a question for Jesus but he's away. I think the intent of the original code is to fail if it isn't passed suitable values for channelId or teamId.

Comment (grundleborg) [reply components/permissions_gates/channel_permission_gate/index.js]: In fact, thinking about it some more, why are you changing this? Is the current team necessarily the correct team?

Comment (enahum) [reply components/permissions_gates/channel_permission_gate/index.js]: @grundleborg  I just did this in order to have the options available to test this PR, currently this functionality is broke on `master` as it seems the component receives an `undefined` teamId somehow.

Now after inspecting the code I did find that this component is only mounted for the current team.

Once this PR is actually tested I'm going to remove this piece of code and leave it to XZY to do the proper fix for the break in master

Comment (grundleborg) [reply components/permissions_gates/channel_permission_gate/index.js]: OK... I wonder how it got broken, as nothing has changed in the permissions gate code recently. I'm not comfortable signing off on this fix without Jesus' opinion, but I'll ask Martin to take a look.

I do wonder why it's only broken now though? Suggests to me that some other change is at fault, not the permissions gates themselves.

Comment (enahum) [reply components/permissions_gates/channel_permission_gate/index.js]: possibly something else broke it, don't know what it was though

**Comment by DHaussermann** [comment]
Regression tested around channel creation and renaming. No issues found.
Also verified bug from previous PR is resolved.

@enahum  Looks good to me!

**Comment by enahum** [comment]
@grundleborg @jwilander now that the PR has been tested I'll remove the changes for the permission stuff and leave it up to the XYZ team

**Comment by enahum** [comment]
@grundleborg @jwilander I've reverted the changes to the permission gate as discussed

**Pull request merged**